diagnostics-extension/src/sw.ts

```diff
@@ -10,18 +10,44 @@ import {
   Request,
   RequestType,
   ResponseErrorInfoMessage,
+  PortName,
 } from '@common/message';
 import { handleDiagnostics } from './controllers/diagnostics';
-import { handleEvents } from './controllers/events';
+import {
+  handleEvents,
+  registerEventHandlers,
+  registerPort,
+} from './controllers/events';
 import { handleTelemetry } from './controllers/telemetry';
 import { generateErrorResponse } from './utils';
 
+let eventHandlerAdded: boolean; // true if the event handlers are already added
+
 chrome.runtime.onInstalled.addListener(
   (details: chrome.runtime.InstalledDetails) => {
     console.log('Service worker is installed!', details);
+    eventHandlerAdded = false;
   }
 );
 
+chrome.runtime.onConnectExternal.addListener((port: chrome.runtime.Port) => {
+  if (port.name === PortName.EVENTS_PORT) {
+    registerPort(port);
+    if (!eventHandlerAdded) {
+      registerEventHandlers().then(response => {
+        if (!response.success) {
+          console.error(response.error?.message);
+        }
+        eventHandlerAdded = true;
+      }).catch(err => {
+        console.error(err.message);
+      });
+    }
```

Comment:
Can we move this all into event.ts?

Maybe a function like `onEventPortConnect`, and inside `events.ts` we can do all the set up. That way we don't need to export `registerPort` and `registerEventHandlers`. 

I think this code should not belong to sw.ts

Reply:
Yeah sure. I added <code>onEventPortConnect</code> in the events controller.